crates/libcontainer/src/workload/default.rs

```diff
@@ -40,8 +41,123 @@ impl Executor for DefaultExecutor {
         // payload through execvp, so it should never reach here.
         unreachable!();
     }
+
+    fn validate(&self, spec: &Spec) -> Result<(), ExecutorError> {
+        let proc = spec
+            .process()
+            .as_ref()
+            .ok_or_else(|| ExecutorError::InvalidArg)?;
+
+        if let Some(args) = proc.args() {
+            let envs: Vec<String> = proc.env().as_ref().unwrap_or(&vec![]).clone();
+            let path_vars: Vec<&String> = envs.iter().filter(|&e| e.starts_with("PATH=")).collect();
+            if path_vars.is_empty() {
+                tracing::error!("PATH environment variable is not set");
+                Err(ExecutorError::InvalidArg)?;
+            }
+            let path_var = path_vars[0].trim_start_matches("PATH=");
+            match get_executable_path(&args[0], path_var) {
+                None => {
+                    tracing::error!(
+                        "executable {} for container process not found in PATH",
+                        args[0]
+                    );
+                    Err(ExecutorError::InvalidArg)?;
+                }
+                Some(path) => match is_executable(&path) {
+                    Ok(true) => {
+                        tracing::debug!("found executable {:?}", path);
+                    }
+                    Ok(false) => {
+                        tracing::error!(
+                            "executable {:?} does not have the correct permission set",
+                            path
+                        );
+                        Err(ExecutorError::InvalidArg)?;
+                    }
+                    Err(err) => {
+                        tracing::error!(
+                            "failed to check permissions for executable {:?}: {}",
+                            path,
+                            err
+                        );
+                        Err(ExecutorError::InvalidArg)?;
+                    }
+                },
+            }
+        }
+
+        Ok(())
+    }
 }
 
 pub fn get_executor() -> Box<dyn Executor> {
     Box::new(DefaultExecutor {})
 }
+
+fn get_executable_path(name: &str, path_var: &str) -> Option<PathBuf> {
+    // if path has / in it, we have to assume absolute path, as per runc impl
+    if name.contains('/') && PathBuf::from(name).exists() {
+        return Some(PathBuf::from(name));
+    }
+    for path in path_var.split(':') {
```

Comment:
I know that this PR just moved code but I want to raise the concern where @jprendes raised here: https://github.com/containerd/runwasi/pull/156#discussion_r1284160469

The issue is that the `is_executable` check should be inside of the for loop, not outside. @jprendes provides an example of illustrating why that's the case.

Reply:
@Mossaka I have created the issue to address what you mentioned https://github.com/containers/youki/issues/2277. I think this issue is outside the scope of this PR so we should take care of it in another PR, but I appreciate you pointed out, thanks.